go-controller/pkg/ovn/ovn.go

```diff
@@ -520,7 +525,7 @@ func (oc *Controller) ensurePod(oldPod, pod *kapi.Pod, addPort bool) bool {
 			return false
 		}
 	} else {
-		if oldPod != nil && exGatewayAnnotationsChanged(oldPod, pod) {
+		if oldPod != nil && (exGatewayAnnotationsChanged(oldPod, pod) || networkStatusAnnotationsChanged(oldPod, pod)) {
```

Comment:
So I guess this condition handles if there is an update to the multus annotation, then delete. I guess looking at the logic now on line 532 we were always adding the exgw anyway? So did this already work, and now we are just fixing the case where the multus annotations would get updated, like an IP changes in a network attachment def?

Reply:
Yes, the code currently handles the multus annotations (network-status) without any changes. I was thinking it would be good to handle the case where the multus annotations changed, which I think is possible.

Reply:
Update: I just had a chat with Doug from the multus team, and he was saying that multus will not change the network-status annotations once the pod has been created.

"But, it won't change by the hand of Multus. It gets created only during a CNI ADD, after being processed by Multus. It happens once, and that's it. There's no updating or changing it by Multus."

Reply:
Another follow up to this conversation. tl;dr: The IP address can be changed after the pod is up, for example, by manually editing the pod and changing the network-status annotation.